lib/streamer/cbor_streamer.go

```diff
@@ -0,0 +1,199 @@
+package streamer
+
+import (
+	"io"
+	"math"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/spacemonkeygo/errors"
+	"github.com/ugorji/go/codec"
+)
+
+var _ Mux = &CborMux{}
+
+type CborMux struct {
+	// the fd tip handles append; reads all use ReadAt and track their offsets individually
+	file  *os.File
+	codec *codec.Encoder
+	wmu   sync.Mutex
+}
+
+type cborMuxRow struct {
+	Label int    `json:"l"`
+	Msg   []byte `json:"m,omitempty"`
+	Sig   int    `json:"x,omitempty"` // 1->closed
+}
+
+func CborFileMux(filePath string) Mux {
+	file, err := os.OpenFile(filePath, os.O_CREATE|os.O_RDWR|os.O_EXCL, 0755)
+	if err != nil {
+		panic(errors.IOError.Wrap(err))
+	}
+	file.Write([]byte{codec.CborStreamArray})
```

Comment:
I'm going to claim that this [spec section](https://tools.ietf.org/html/rfc7049#section-2.2.1) appears to give us clearance to not have a terminus. I would feel better if we had one though - it feels... more competent? ... to be able to definitely know when we're done. Will probably also be useful if these streams end up getting sent over something with a more complex EOF reality (network).